routing/cross_mwm_index_graph.cpp

```diff
@@ -174,4 +291,84 @@ void CrossMwmIndexGraph::InsertWholeMwmTransitionSegments(NumMwmId numMwmId)
   if (!LoadWith(numMwmId, fillAllTransitionSegments))
     m_transitionCache.emplace(numMwmId, TransitionSegments());
 }
+
+void CrossMwmIndexGraph::GetBorderCross(TRoutingMappingPtr const & mapping, Segment const & s,
+                                        bool isOutgoing, vector<BorderCross> & borderCrosses)
+{
+  // ingoing edge
+  NodeIds const nodeIdsTo = GetDirectAndReverseNodeId(mapping->m_segMapping, s);
+
+  vector<ms::LatLon> const & transitionPnt =
+      isOutgoing ? GetIngoingTransitionPnt(s) : GetOutgoingTransitionPnt(s);
+  CHECK(!transitionPnt.empty(), ("Segment:", s, ", isOutgoing:", isOutgoing));
+
+  // If |isOutgoing| == true |nodes| is "to" cross nodes, otherwise |nodes| is "from" cross nodes.
+  vector<CrossNode> nodes;
+  for (ms::LatLon const & p : transitionPnt)
+    nodes.emplace_back(nodeIdsTo.m_directNodeId, nodeIdsTo.m_reverseNodeId, mapping->GetMwmId(), p);
+
+  // outgoing edge
+  vector<Segment> twins;
+  GetTwins(s, !isOutgoing, twins);
+  CHECK(!twins.empty(), ("Segment:", s, ", isOutgoing:", isOutgoing));
```

Comment:
CrossMwmIndexGraph::GetBorderCross(...) is a private method. It's called (It is written to be called) only from CrossMwmIndexGraph::GetEdgeList(...). If s ingoing and outgoing at the same time there's an early return in CrossMwmIndexGraph::GetEdgeList(...) and CrossMwmIndexGraph::GetBorderCross(...) is not called.